content/4_Gold/MST.mdx

```diff
@@ -144,64 +144,61 @@ no edges are counted twice, we cannot "accidentally" count $n-1$ edges.
 <CPPSection>
 
 ```cpp
-#include <iostream>
-#include <vector>
-#include <algorithm>
+#include <bits/stdc++.h>
 using namespace std;
 
-typedef long long ll;
-typedef pair<int, int> pi;
-typedef vector<int> vi;
-
-#define trav(a,x) for (auto& a: x)
-#define all(x) begin(x), end(x)
-#define mp make_pair
-#define f first
-#define s second
-
 struct DSU {
-	vi e; void init(int N) { e = vi(N,-1); }
-	// get representive component, uses path compression
+	vector<int> e; 
+	DSU(int N) : e(N, -1) { }
 	int get(int x) { return e[x] < 0 ? x : e[x] = get(e[x]); }
-	bool sameSet(int a, int b) { return get(a) == get(b); }
+	bool connected(int a, int b) { return get(a) == get(b); }
 	int size(int x) { return -e[get(x)]; }
-	bool unite(int x, int y) { // union by size
-		x = get(x), y = get(y); if (x == y) return 0;
-		if (e[x] > e[y]) swap(x,y);
-		e[x] += e[y]; e[y] = x; return 1;
+	bool unite(int x, int y) { 
+		x = get(x), y = get(y); 
+		if (x == y) {
+			return false;
+		}
+		if (e[x] > e[y]) {
+			swap(x,y);
+		}
+		e[x] += e[y]; 
+		e[y] = x; 
+		return true;
 	}
 };
 
-int cnt=0;
-template<class T> T kruskal(int N, vector<pair<T,pi>> ed) {
-	sort(all(ed));
-	T ans = 0; DSU D; D.init(N+1); // edges that unite are in MST
-	trav(a,ed)
-	if (D.unite(a.s.f,a.s.s)) {
-		ans += a.f;
-		cnt++;
+template<class T> T kruskal(int N, vector<pair<T, pair<int, int>>> edges) {
```

Comment:
why the generic?

Reply:
it was generic and I didnt change it, don't think this is too big an issue